attest/win_events.go

```diff
@@ -394,13 +409,93 @@ eventLoop:
 	return nil
 }
 
+// parseUTF16 decodes data representing a UTF16 string. It is assumed the
+// caller has validated that the data size is within allowable bounds.
+func (w *WinEvents) parseUTF16(header microsoftEventHeader, r io.Reader) (string, error) {
+	data := make([]uint16, header.Size/2)
+	if err := binary.Read(r, binary.LittleEndian, &data); err != nil {
+		return "", err
+	}
+	return strings.TrimSuffix(string(utf16.Decode(data)), string(0x00)), nil
+}
+
+func (w *WinEvents) readELAMAggregation(rdr *bytes.Reader, header microsoftEventHeader) error {
+	var (
+		r          = io.LimitReader(rdr, int64(header.Size))
+		driverName string
+		measured   []byte
+		policy     []byte
+		config     []byte
+	)
+
+eventLoop:
+	for {
+		var header microsoftEventHeader
+		if err := binary.Read(r, binary.LittleEndian, &header); err != nil {
+			if errors.Is(err, io.EOF) {
+				break eventLoop
+			}
+			return fmt.Errorf("parsing ELAM aggregation sub-event: %v", err)
+		}
+		if int(header.Size) > rdr.Len() {
```

Comment:
This should be r.N instead of rdr.Len(), right? Since reading more than the limited reader will produce an error below.